rules/aip0132/resource_reference_type_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0132
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestResourceReferenceType(t *testing.T) {
+	// Set up testing permutations.
+	tests := []struct {
+		testName string
+		TypeName string
+		RefType  string
+		problems testutils.Problems
+	}{
+		{"ValidChildType", "library.googleapis.com/Book", "child_type", nil},
+		{"ValidType", "library.googleapis.com/Shelf", "type", nil},
```

Comment:
How does the lint rule know that the parent type is Shelf? That string does not appear in the example Message. It makes me wonder if it is actually being validated--the the `type` (wrt `resource_reference`) is, in fact correct, `library.googleapis.com/Shelf`. I *think* that we are just checking that it is not the the `child_type`. In turn, it is not really valid, but it is not, not valid, iiuc.

Reply:
You're right, this is a naive check, it doesn't go as far as comparing each `pattern` in the child type to those in the parent type and ensuring things line up. Do you think that belongs in this check or in a different one? It only applies to the direct `type` reference.

Reply:
I wonder if we should even allow use of `type` anymore...`child_type` is the strictly better reference type for these RPCs as it is "forward compatible" without needing to change anything. If a new parent pattern is added to a resource, and the parent fields are using a direct `type` reference, they would need to be updated to child_type in order to enable use of the new parent pattern for that resource e.g. having a resource parented by just project, and then adding organization as a parent pattern.

WDYT?

Reply:
SGTM. I wonder how many references there are that use `type. ?